Doc/tools/rstlint.py

```diff
@@ -128,6 +129,66 @@ def check_leaked_markup(fn, lines):
             yield lno+1, 'possibly leaked markup: %r' % line
 
 
+def hide_literal_blocks(lines):
+    """Tool to remove literal blocks from given lines.
+
+    It yields empty lines in place of blocks, so line numbers are
+    still meaningfull.
+    """
+    in_block = False
+    for line in lines:
+        if line.endswith("::\n"):
+            in_block = True
+        elif in_block:
+            if line == "\n" or line.startswith(" "):
+                line = "\n"
+            else:
+                in_block = False
+        yield line
+
+def hide_comments(lines):
+    """Tool to remove comments from given lines.
+
+    It yields empty lines in place of comments, so line numbers are
+    still meaningfull.
+    """
+    in_multiline_comment = False
+    for line in lines:
+        if line == "..\n":
+            in_multiline_comment = True
+        elif in_multiline_comment:
+            if line == "\n" or line.startswith(" "):
+                line = "\n"
+            else:
+                in_multiline_comment = False
+        if line.startswith(".. "):
```

Comment:
Sorry if I'm misinterpreting something here, but wouldn't this remove directives like `.. describes::` or `.. class::` too? Or is that intentional?

Reply:
You're understanding it correctly, it removes a bit too much. But the documentation tells comments are `Every explicit markup block which isn’t a valid markup construct`.

an `explicit markup` is:

> An explicit markup block begins with a line starting with .. followed by whitespace and is terminated by the next paragraph at the same level of indentation.

According to the [ref](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#explicit-markup-blocks), explicite markup are

> used for footnotes, citations, hyperlink targets, directives, substitution definitions, and comments

in `directives` there's potentially a lot, as it's possible to add new directives, like `versionchanged` in our doc.

So we have like three choices:
 - Keep it simple and introduce some false negatives (current version)
 - Do it right with no false positives and no false negative, it's probably doable as we have the exhaustive list of directives already hardcoded in rstlint.py
 - Keep it simple, the other way around, and introduce some false positives.

I'm really not in favor of false positives: it's a burden for contributors. False negatives are OK though: if an rst error is introduced it'll still fixable later.

I'll try myself at the "do it right version" today to see if it's worth it.

Reply:
Was not that hard, probably worth it, I pushed it.

Reply:
Thanks for your patience and explaining it to me :).